tests/test_utils.py

```diff
@@ -34,21 +34,23 @@ def bootstrap_config():
 def assert_exception_correct(got: Exception, expected: Exception):
     err = "".join(traceback.TracebackException.from_exception(got).format())
     assert got.args == expected.args, err
-    assert type(got) == type(expected)
+    assert type(got) == type(expected)  # noqa: E721
```

Comment:
I presume this rule wants you to use `isinstance` instead of type? Better to remove the `# noqa` and either fix it in this PR or in an upcoming one since it's a simple fix.

Reply:
I believe I had discussions with @MrCreosote about how it needed to be this way rather than isinstance, but maybe I am mis-remembering. Can you chime in here @MrCreosote ?

Reply:
Because you want to assert against the exact exception type, not the type's inheritance hierarchy

Reply:
Oh -- in that case I'm sure that there are ways around this.

Reply:
What do I need to do next?

Reply:
For now, remove `# noqa`, it can be fixed in an upcoming PR (I am happy to fix it once you have finished your PRs).

Reply:
As we discussed in slack, we are keeping this one ` # noqa: E721` and changing the rest of them to be isinstance

Reply:
Changed the rest of the isinstance